From d745f8d7148329fb8567bd337ec9c3e83977db12 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:55 -0400
Subject: [usb] host/xhci-plat: add xhci_plat_start()

Message-id: <1410887463-20674-524-git-send-email-dzickus@redhat.com>
Patchwork-id: 95624
O-Subject: [RHEL7 PATCH 523/591] usb: host: xhci-plat: add xhci_plat_start()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 94adcdce0d340e06eb9187cad70555d2e8a201f3
Author: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
Date:   Wed May 28 20:22:58 2014 +0900

    usb: host: xhci-plat: add xhci_plat_start()

    Some platforms (such as the Renesas R-Car) need to initialize some specific
    registers after xhci driver calls usb_add_hcd() and before the driver calls
    xhci_run(). So, this patch adds the xhci_plat_start() function.

    Acked-by: Geert Uytterhoeven <geert@linux-m68k.org>
    Signed-off-by: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 3fce2c7..30a46bb 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -36,6 +36,11 @@ static int xhci_plat_setup(struct usb_hcd *hcd)
  return xhci_gen_setup(hcd, xhci_plat_quirks);
 }
 
+static int xhci_plat_start(struct usb_hcd *hcd)
+{
+ return xhci_run(hcd);
+}
+
 static const struct hc_driver xhci_plat_xhci_driver = {
  .description =  "xhci-hcd",
  .product_desc =  "xHCI Host Controller",
@@ -51,7 +56,7 @@ static const struct hc_driver xhci_plat_xhci_driver = {
   * basic lifecycle operations
   */
  .reset =  xhci_plat_setup,
- .start =  xhci_run,
+ .start =  xhci_plat_start,
  .stop =   xhci_stop,
  .shutdown =  xhci_shutdown,
 
-- 
1.7.1